internal/nginx/config/servers_test.go

```diff
@@ -200,6 +200,19 @@ func TestCreateServers(t *testing.T) {
 							},
 						},
 					},
+					Filters: []v1beta1.HTTPRouteFilter{
```

Comment:
I don't think it is necessary to add the RequestHeaderModifier to multiple places. 
Instead,I suggest adding one new match like below:
```
				{
					// A match with a redirect with implicit port
					Matches: []v1beta1.HTTPRouteMatch{
						{
							Path: &v1beta1.HTTPPathMatch{
								Value: helpers.GetStringPointer("/redirect-implicit-port"),
								Type:  helpers.GetPointer(v1beta1.PathMatchPathPrefix),
							},
						},
					},
					// redirect is set in the corresponding state.MatchRule
				},
	```